lib/json/api/relationship.rb

```diff
@@ -7,11 +7,12 @@ class Relationship
       def initialize(relationship_hash, options = {})
         @options = options
         @links_defined = relationship_hash.key?('links')
+        @data_defined = relationship_hash.key?('data')
+        @meta_defined = relationship_hash.key?('meta')
         links_hash = relationship_hash['links'] || {}
         @links = Links.new(links_hash, @options)
-        @data = parse_linkage(relationship_hash['data']) if
-          relationship_hash.key?('data')
-        @meta = relationship_hash['meta'] if relationship_hash.key?('meta')
+        @data = parse_linkage(relationship_hash['data']) if @data_defined
+        @meta = relationship_hash['meta'] if @meta_defined
```

Comment:
Could you leave stuff in the same order it was before, so that it minimises changes in this PR?